wolfcrypt/src/wolfmath.c

```diff
@@ -106,4 +118,132 @@ int mp_rand(mp_int* a, int digits, WC_RNG* rng)
 }
 #endif /* WC_RSA_BLINDING */
 
+
+#ifdef HAVE_WOLF_BIGINT
+void wc_bigint_init(WC_BIGINT* a)
+{
+    if (a) {
+        a->buf = NULL;
+        a->len = 0;
+        a->heap = NULL;
+    }
+}
+
+int wc_bigint_alloc(WC_BIGINT* a, word32 sz)
+{
+    int err = MP_OKAY;
+
+    if (a == NULL)
+        return BAD_FUNC_ARG;
+
+    if (sz > 0) {
+        if (a->buf && sz > a->len) {
+            wc_bigint_free(a);
+        }
+        if (a->buf == NULL) {
```

Comment:
I think this would look cleaner:
  if (a->buf == NULL) {
      a->buf = ...XMALLOC...
  }
  if (a->buf == NULL) {
      err = MP_MEM;
  }
  else {
      XMEMSET...
  }

Just an idea.